### ISSUE TITLE
Authentication using fingerprint inside applications

### ISSUE THREAD

**Issue by mihaelmilea** [ubports/ubports-touch#221]
- Device: Turbo
- Channel: Any
- Build: Any

## Steps to reproduce

1. Open terminal
2. The passcode is required

## Expected behavoir

It would be great to be able to authenticate with the fingerprint.

## Actual behavior

I have to enter passcode for terminal, file manager, 
UT Tweak Tool, etc.

## Logfiles and additional information

...

**Comment by NeoTheThird** [comment]
This issue was moved to ubports/terminal-app#8

**Comment by NeoTheThird** [comment]
Thanks for reporting :) Moved this to the terminal app issue tracker :)

**Comment by ChloeWolfieGirl** [comment]
Question, can you have a password lock for you lockscreen, and use your finger print to unlock apps?

**Comment by NeoTheThird** [comment]
@ChloeWolfieGirl You can't use the fingerprint to unlock apps, that's what this issue is about.

**Comment by sverzegnassi** [comment]
Using the registered fingerprint for authenticating inside "root" applications is definitely desirable.
Talking about File Manager and Terminal, the password is currently required in order to prevent the access from unauthorized users: using the fingerprint (where available) might be convenient as it makes the operation faster.
Talking about UT Tweak Tool, the password is required in order to execute commands with 'sudo': in this case, I believe that a fingerprint would hardly fit with the app requirements. However, the password request could be postponed until the user performs an action that requires a sudo access.

What UBports could do, in order to make easier for developers to enforce a secure access to their 'unconfined' apps, is to provide a QML module that handles authentication in a transparent manner.
If someone wants to develop e.g. a code editor that can access the system files, we should try to make easier for him to enforce proper security policies.
As for all the non-trivial topics, the question is actually related to a bigger picture, and a bug tracker might not be the best medium for a proper discussion. @NeoTheThird, could we follow up this conversation at UbuCon? Would it be okay for you?

**Comment by NeoTheThird** [comment]
@sverzegnassi Yep, sounds good. Let's talk at UbuCon.

**Comment by ChloeWolfieGirl** [comment]
I typed that when the issue was closed, but seems to have been reopened between the time I opened the page and the time I sent the message.

**Comment by NeoTheThird** [comment]
@ChloeWolfieGirl Yeah, sorry about that. I moved the issue at first, but that was wrong.

**Comment by kissstudio** [comment]
how is it going now? Can we have the app sdk with fingerprint support?
